syntax = "proto3";

package pb;


service CS {
    // 加入对局
    rpc JoinPlayer(CSReqJoinPlayer) returns (CSResJoinPlayer) {}
    // 开始对局
    rpc StartGameplay(CSReqStartGameplay) returns (CSResStartGameplay) {}
    // call
    rpc Call(CSReqCall) returns (CSResCall) {}
    // raise
    rpc Raise(CSReqRaise) returns (CSResRaise) {}
    // fold
    rpc Fold(CSReqFold) returns (CSResFold) {}
    // check
    rpc Check(CSReqCheck) returns (CSResCheck) {}
}

service SC {
    // 同步单局信息
    rpc SyncGameplay(CSReqSyncGameplay) returns (CSResSyncGameplay) {}
}

message CSGameplay {
    repeated Player players = 1; // 玩家信息
    int32 button_player_index = 2; // 庄家位置下标

    message Player {
        uint64 uid = 1; // 玩家id
        int32 index = 2; // 位置
    }
}

message CSReqSyncGameplay {
    uint64 uid = 1; // 玩家id
    CSGameplay gameplay = 2;
}

message CSResSyncGameplay {}

message CSReqJoinPlayer {
    uint64 uid = 1; // 玩家id
    string name = 2; // 名字
    int64 chips = 3; // 自己的筹码
    string ip = 5; // 自己的ip
    uint32 port = 4; // 自己的端口
}

message CSResJoinPlayer {}

message CSReqStartGameplay {
    uint64 uid = 1; // 目前谁开始都可以
}

message CSResStartGameplay {}

message CSReqCall {
    uint64 uid = 1; // 玩家id
}

message CSResCall {}

message CSReqCheck {
    uint64 uid = 1; // 玩家id
}

message CSResCheck {}

message CSReqFold {
    uint64 uid = 1; // 玩家id
}

message CSResFold {}

message CSReqRaise {
    uint64 uid = 1; // 玩家id
}

message CSResRaise {}